mods/magnifier-headless.wh.cpp

```diff
@@ -0,0 +1,172 @@
+    if (hwnd) {
+        ShowWindow(hwnd, SW_HIDE);
+    }
+    
+    // Find by ScreenMagnifierUIWnd
+    hwnd = FindWindowW(L"ScreenMagnifierUIWnd", NULL);
+    if (hwnd) {
+        ShowWindow(hwnd, SW_HIDE);
+    }
+}
+
+// Monitoring thread with proper shutdown
+DWORD WINAPI MonitorThread(LPVOID lpParam) {
+    while (!g_shouldStop) {
+        HideMagnifierWindows();
+        
+        // Sleep with early exit check
+        for (int i = 0; i < 50 && !g_shouldStop; i++) {
+            Sleep(10); // Total 500ms but can exit early
+        }
+    }
+    return 0;
+}
+
+HANDLE g_thread = NULL;
+
+// ShowWindow hook to catch magnifier show attempts
+using ShowWindow_t = decltype(&ShowWindow);
+ShowWindow_t ShowWindow_Original;
+
+BOOL WINAPI ShowWindow_Hook(HWND hWnd, int nCmdShow) {
+    // If it's a magnifier window trying to show, hide it instead
+    if (IsMagnifierWindow(hWnd) && 
+        (nCmdShow == SW_SHOW || nCmdShow == SW_RESTORE || nCmdShow == SW_MAXIMIZE)) {
+        return ShowWindow_Original(hWnd, SW_HIDE);
+    }
+    
+    return ShowWindow_Original(hWnd, nCmdShow);
+}
+
+// CreateWindowExW hook to catch magnifier window creation
+using CreateWindowExW_t = decltype(&CreateWindowExW);
+CreateWindowExW_t CreateWindowExW_Original;
+
+HWND WINAPI CreateWindowExW_Hook(
+    DWORD dwExStyle,
+    LPCWSTR lpClassName,
+    LPCWSTR lpWindowName,
+    DWORD dwStyle,
+    int X,
+    int Y,
+    int nWidth,
+    int nHeight,
+    HWND hWndParent,
+    HMENU hMenu,
+    HINSTANCE hInstance,
+    LPVOID lpParam
+) {
+    HWND hwnd = CreateWindowExW_Original(
+        dwExStyle, lpClassName, lpWindowName, dwStyle,
+        X, Y, nWidth, nHeight, hWndParent, hMenu, hInstance, lpParam
+    );
+    
+    // If created window is magnifier, hide it immediately
+    if (hwnd && lpClassName) {
+        if (wcscmp(lpClassName, L"MagUIClass") == 0 ||
+            wcscmp(lpClassName, L"ScreenMagnifierUIWnd") == 0) {
+            ShowWindow(hwnd, SW_HIDE);
+        }
+    }
+    
+    return hwnd;
+}
+
+// Mod initialization
+BOOL Wh_ModInit() {
+    g_shouldStop = FALSE;
+    
+    // Initial hide
+    HideMagnifierWindows();
+    
+    // Create monitoring thread
+    g_thread = CreateThread(NULL, 0, MonitorThread, NULL, 0, NULL);
+    
+    return TRUE;
+}
+
+// Proper cleanup without TerminateThread
+void Wh_ModUninit() {
+    // Signal thread to stop
+    g_shouldStop = TRUE;
+    
+    // Wait for thread to finish properly
+    if (g_thread) {
+        WaitForSingleObject(g_thread, 2000); // Wait max 2 seconds
+        CloseHandle(g_thread);
+        g_thread = NULL;
+    }
+    
+    // Restore magnifier windows by class name
+    HWND hwnd = FindWindowW(L"MagUIClass", NULL);
+    if (hwnd) {
+        ShowWindow(hwnd, SW_SHOW);
+    }
+    
+    hwnd = FindWindowW(L"ScreenMagnifierUIWnd", NULL);
+    if (hwnd) {
+        ShowWindow(hwnd, SW_SHOW);
+    }
+}
+
+// Hook setup
+BOOL Wh_ModSettingsInit() {
```

Comment:
Such function is also not part of Windhawk's API. The documentation can be found here:
https://github.com/ramensoftware/windhawk/wiki/Creating-a-new-mod